### PR TITLE
refactor: refactored types and composables to new ones from theme

### PR DESCRIPTION
**Pull request by Frodigo**
## Description
- refactored CurrenciesModal to use new composable 
- refactored StoreSwitcher to use new composables
- refactored other components to use new composables
- refactored getters to use types from theme

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/theme/composables/types.d.ts [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import { CustomQuery } from '@vue-storefront/core/lib/src/types';
+import {FacetSearchResult} from "@vue-storefront/core";
```
Comment (sethidden) [inline packages/theme/composables/types.d.ts:2]: Mind copying this over from core? This is probably some one liner

File: packages/theme/composables/types.d.ts [hunk 2]
```diff
@@ -51444,3 +51451,55 @@ export declare type WishlistQuery = {
     } | null | undefined>;
   } | null | undefined;
 };
+
+export declare type Category = CategoryTree | CategorySearchQuery['categoryList'][0];
```
Comment (sethidden) [inline packages/theme/composables/types.d.ts:51455]: Do these come from re-running some generation script? i.e. those + diffs are not manually added, right?

Comment (Frodigo) [reply packages/theme/composables/types.d.ts:51455]: Yes they are auto-generated.

File: packages/theme/composables/types.d.ts (hunk 1 above)
Comment (Frodigo) [reply packages/theme/composables/types.d.ts:2]: Done.

**Review by sethidden — APPROVED** [review]

**Pull request merged**